pandas/tests/series/methods/test_replace.py

```diff
@@ -557,6 +557,14 @@ def test_pandas_replace_na(self):
         exp = pd.Series(["CC", "CC", "CC-REPL", "DD", "CC", "", pd.NA], dtype="string")
         tm.assert_series_equal(result, exp)
 
+    def test_replace_dictlike_value_None(self):
+        # GH46004
+        ser = pd.Series([1, 2, 3, 4])
+        result = ser.replace(to_replace={3: 33}, value=None)
```

Comment:
I don't think this is supposed to work. We are using lib.no_default as a sentinel to check if value was provided.

Reply:
It seems like when we have explicitly to_replace=dict_like  and value=None it interpreted as a special case where None is different from lib.no_default. Maybe we need another conditional statement somewhere.